### ISSUE TITLE
Various improvements for GH actions

### ISSUE THREAD

**Issue by christian-kreuzberger-dtx** [keptn/keptn#2824]
This issue is a follow-up issue for PR #2817 . In addition, it acts as a meta-issue for making sure the GH Actions CI workflow is working as expected.

Plus:
* [x] Automatically create a dev (or latest) release which stores the CLI and Helm-Chart such that it can be directly downloaded
* [x] Automatically create a release draft for changes on release branches
* [x] Move certain "bash scripts" into a separate re-usable (for keptn-contrib/sandbox) GH action
* [x] Enable MacOS builds again once maintenance is complete
* [x] Upgrade next-semvers action to allow pre-releases https://github.com/WyriHaximus/github-action-next-semvers/releases/tag/v1.1.0
